### ISSUE TITLE
[QST] Activation Checkpointing

### ISSUE THREAD

**Issue by jeromeku** [pytorch/torchtune#2403]
A few questions for activation checkpointing:

1) Why are `use_reentrant` and `preserve_rng_state` hardcoded to `False`?
2) How to do more finegrained selective checkpointing as in this [paper](https://arxiv.org/abs/2205.05198)?  Currently either only full or every `x` layer checkpointing is supported.

Thanks!

**Comment by jeromeku** [comment]
@ebsmothers 

Many thanks for the comprehensive response!  Will dig into the provided references.

Amazed by the progress of torchtune since I last checked in!

Just wrapped up a large project and am looking to jump back into contributing OSS.  Are there particular areas of `torchtune` that could use some help?  

Given renewed interest in RL as of late, would be interested to see how pytorch 2.0 stack (`compile`, `ao`, `DTensor`) can be composed to optimize these multi-stage workflows (generation, ref model / reward , actor training).  Will look over the DPO recipe as well as the GRPO draft and compare with existing RL-specific post-training frameworks.  Also will take a look at the `MoE` RFC.

**Comment by ebsmothers** [comment]
@jeromeku fyi I am going through our recent issues and cleaning up anything that's already addressed so going to close this one. Feel free to reopen if you still have more questions on activation checkpointing!
